# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/sandbox/chromium/sandbox/win/src/acl.cc
# Commit: 370c40e49230
# Full Hash: 370c40e4923073a87845b57c1f4851bacd9663da
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2019-06-12 21:42:35
# Regressor Bug: 1552160
# File Overlap Count: 2
# Description:
#   Bug 1552160 Part 1: Roll-up of chromium sandbox update and mozilla patches to get a running browser. r=jld,aklotz,tjr,bobowen
#   
#   This updates security/sandbox/chromium/ files to chromium commit 84108231f6e6e0772fb9a4643679ce76aa771e67.
#   
#   Existing and new patches applied from security/sandbox/chromium-shim/patches/with_update/ to give a compiling and mostly working browser.
# ==============================================================================

diff -r bd5ecd72f76c -r 370c40e49230 security/sandbox/chromium/sandbox/win/src/acl.cc
--- a/security/sandbox/chromium/sandbox/win/src/acl.cc	Wed Jun 12 21:55:59 2019 +1200
+++ b/security/sandbox/chromium/sandbox/win/src/acl.cc	Wed Jun 12 11:10:48 2019 +0100
@@ -15,13 +15,13 @@
 bool GetDefaultDacl(
     HANDLE token,
     std::unique_ptr<TOKEN_DEFAULT_DACL, base::FreeDeleter>* default_dacl) {
-  if (token == NULL)
+  if (!token)
     return false;
 
-  DCHECK(default_dacl != NULL);
+  DCHECK(default_dacl);
 
   unsigned long length = 0;
-  ::GetTokenInformation(token, TokenDefaultDacl, NULL, 0, &length);
+  ::GetTokenInformation(token, TokenDefaultDacl, nullptr, 0, &length);
   if (length == 0) {
     NOTREACHED();
     return false;
@@ -33,23 +33,25 @@
 
   if (!::GetTokenInformation(token, TokenDefaultDacl, default_dacl->get(),
                              length, &length))
-      return false;
+    return false;
 
   return true;
 }
 
-bool AddSidToDacl(const Sid& sid, ACL* old_dacl, ACCESS_MODE access_mode,
-                  ACCESS_MASK access, ACL** new_dacl) {
+bool AddSidToDacl(const Sid& sid,
+                  ACL* old_dacl,
+                  ACCESS_MODE access_mode,
+                  ACCESS_MASK access,
+                  ACL** new_dacl) {
   EXPLICIT_ACCESS new_access = {0};
   new_access.grfAccessMode = access_mode;
   new_access.grfAccessPermissions = access;
   new_access.grfInheritance = NO_INHERITANCE;
 
-  new_access.Trustee.pMultipleTrustee = NULL;
+  new_access.Trustee.pMultipleTrustee = nullptr;
   new_access.Trustee.MultipleTrusteeOperation = NO_MULTIPLE_TRUSTEE;
   new_access.Trustee.TrusteeForm = TRUSTEE_IS_SID;
-  new_access.Trustee.ptstrName = reinterpret_cast<LPWSTR>(
-                                    const_cast<SID*>(sid.GetPSID()));
+  new_access.Trustee.ptstrName = reinterpret_cast<LPWSTR>(sid.GetPSID());
 
   if (ERROR_SUCCESS != ::SetEntriesInAcl(1, &new_access, old_dacl, new_dacl))
     return false;
@@ -61,14 +63,14 @@
                          const Sid& sid,
                          ACCESS_MODE access_mode,
                          ACCESS_MASK access) {
-  if (token == NULL)
+  if (!token)
     return false;
 
   std::unique_ptr<TOKEN_DEFAULT_DACL, base::FreeDeleter> default_dacl;
   if (!GetDefaultDacl(token, &default_dacl))
     return false;
 
-  ACL* new_dacl = NULL;
+  ACL* new_dacl = nullptr;
   if (!AddSidToDacl(sid, default_dacl->DefaultDacl, access_mode, access,
                     &new_dacl))
     return false;
@@ -76,10 +78,10 @@
   TOKEN_DEFAULT_DACL new_token_dacl = {0};
   new_token_dacl.DefaultDacl = new_dacl;
 
-  BOOL ret = ::SetTokenInformation(token, TokenDefaultDacl, &new_token_dacl,
+  bool ret = ::SetTokenInformation(token, TokenDefaultDacl, &new_token_dacl,
                                    sizeof(new_token_dacl));
   ::LocalFree(new_dacl);
-  return (TRUE == ret);
+  return ret;
 }
 
 bool RevokeLogonSidFromDefaultDacl(HANDLE token) {
@@ -117,26 +119,28 @@
                              GRANT_ACCESS, access);
 }
 
-bool AddKnownSidToObject(HANDLE object, SE_OBJECT_TYPE object_type,
-                         const Sid& sid, ACCESS_MODE access_mode,
+bool AddKnownSidToObject(HANDLE object,
+                         SE_OBJECT_TYPE object_type,
+                         const Sid& sid,
+                         ACCESS_MODE access_mode,
                          ACCESS_MASK access) {
-  PSECURITY_DESCRIPTOR descriptor = NULL;
-  PACL old_dacl = NULL;
-  PACL new_dacl = NULL;
+  PSECURITY_DESCRIPTOR descriptor = nullptr;
+  PACL old_dacl = nullptr;
+  PACL new_dacl = nullptr;
 
-  if (ERROR_SUCCESS != ::GetSecurityInfo(object, object_type,
-                                         DACL_SECURITY_INFORMATION, NULL, NULL,
-                                         &old_dacl, NULL, &descriptor))
+  if (ERROR_SUCCESS !=
+      ::GetSecurityInfo(object, object_type, DACL_SECURITY_INFORMATION, nullptr,
+                        nullptr, &old_dacl, nullptr, &descriptor))
     return false;
 
-  if (!AddSidToDacl(sid.GetPSID(), old_dacl, access_mode, access, &new_dacl)) {
+  if (!AddSidToDacl(sid, old_dacl, access_mode, access, &new_dacl)) {
     ::LocalFree(descriptor);
     return false;
   }
 
-  DWORD result = ::SetSecurityInfo(object, object_type,
-                                   DACL_SECURITY_INFORMATION, NULL, NULL,
-                                   new_dacl, NULL);
+  DWORD result =
+      ::SetSecurityInfo(object, object_type, DACL_SECURITY_INFORMATION, nullptr,
+                        nullptr, new_dacl, nullptr);
 
   ::LocalFree(new_dacl);
   ::LocalFree(descriptor);